Update blender exporter settings

puck7744 opened the pull request:
- Added seemingly missing mixin class for settings panel
- Removed the export active layers option
- Added an export scope option
  - "All" option to export everything in the Blender scene, similar to previous functionality
  - "Selected" option to export only selected objects
  - "Visible" option which replaces the active layer option and also works with Blender's hide/show feature

Palmer-JC commented:
Per the forum thread, the default place with labels pointing to location preserves savability.
` See Scene Properties`
`For Savable Exporter`
`Properities.`

The expansion of selection is an improvement, but think the active layers still needs to be added the list.  Reason is that hide/show is used to export things that import which are also not initially displayed.

puck7744 commented:
Ah, that makes sense! I'll see if I can update the request. :)

Vinc3r commented:
Thanks for this PR, it's a nice improvement :)

deltakosh commented:
@Palmer-JC  ok to merge?

puck7744 commented:
I updated the request to fix some issues with the way I had changed the exporter in line with what @Palmer-JC said.

I also added a simple message to the export settings panel so that it's clear that the lack of ssettings when exporting is not a bug.

Finally, just for my own sanity I changed the `bl_idname` variable to fit Blender's format and thus allow a shortcut to be set for the exporter.

puck7744 commented:
If you do decide to merge, feel free to close #198 as I think it's quite well taken care of 😃

Palmer-JC commented:
Done for the day, will look at this tomorrow

On Wed, May 30, 2018, 6:53 PM Tristan Shelton <notifications@github.com>
wrote:

> If you do decide to merge, feel free to close #198
> <https://github.com/BabylonJS/Exporters/issues/198> as I think it's quite
> well taken care of 😃
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/BabylonJS/Exporters/pull/199#issuecomment-393345319>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHjKY-UPAhP8aGEPZVm1UKM3G5TGXrZ_ks5t3yLlgaJpZM4UOkSa>
> .
>

Palmer-JC commented:
Will generate a zip, after upping version number, & nuking the unused isInSelectedLayer function.  Probably today.  Thanks!

tristan-shelton commented:
@Palmer-JC I actually did re-use isInSelectedLayer here: https://github.com/BabylonJS/Exporters/pull/199/files#diff-dc6dd5305e467a57d4953ed20abe6f7dR85

Looking at that section again I also see that I made a mistake in that it should be operating on `scene.objects` not visible objects. You'll probably want to fix that up before making a release. Sorry about that!

To be clear, line 86 should be `scene.objects` not `bpy.context.visible_objects`
